src/app/home-page/widget/organization-box/organization-box.component.ts

```diff
@@ -38,45 +39,22 @@ export class OrganizationBoxComponent extends Base implements OnInit {
 
   ngOnInit(): void {
     this.getMyOrganizations();
-    this.requestsQuery.myPendingOrganizationRequests$.subscribe(
-      (requests) => {
-        if (requests) {
-          this.pendingRequests = requests;
-        }
-      },
-      (error: HttpErrorResponse) => {
-        this.alertService.detailedError(error);
-      }
-    );
-    this.requestsQuery.myOrganizationInvites$.subscribe(
-      (invites) => {
-        if (invites) {
-          this.pendingRequests = invites;
-        }
-      },
-      (error: HttpErrorResponse) => {
-        this.alertService.detailedError(error);
-      }
-    );
+    this.pendingRequests$ = this.requestsQuery.myPendingOrganizationRequests$;
+    this.pendingInvites$ = this.requestsQuery.myOrganizationInvites$;
   }
 
   private getMyOrganizations() {
     this.usersService
-      .getUserDockstoreOrganizations()
+      .getUserDockstoreOrganizations(null, this.filterText)
       .pipe(
         finalize(() => (this.isLoading = false)),
         takeUntil(this.ngUnsubscribe)
       )
       .subscribe(
         (myOrgs: Array<OrganizationUpdateTime>) => {
-          this.listOfOrganizations = [];
-          myOrgs.forEach((org: OrganizationUpdateTime) => {
-            if (this.listOfOrganizations.length < 5) {
-              this.listOfOrganizations.push(org);
-            }
-            if (this.firstCall) {
-              this.totalOrgs += 1;
-            }
+          myOrgs.forEach(() => {
```

Comment:
You don't need the `forEach()` at all, the body currently just repeats the same operation each time.

Reply:
Yeah no I agree the `forEach()` is redundant, I was following the other dashboard boxes for consistency and couldn't understand why the `forEach()` was there...:
https://github.com/dockstore/dockstore-ui2/blob/b7ddce0c20b7c4991750cfccd039dbe225a91d74/src/app/home-page/widget/entry-box/entry-box.component.ts#L93-L103

I couldn't explain why at the time but when I removed the `forEach()` the search box would go blank when no results returned (From your previous comment I see its because `this.totalOrgs` is set to zero!!):

https://user-images.githubusercontent.com/97123241/219804286-6c4d4cd1-7ee7-4703-a0a4-8d59fa8a4395.mp4

However, with the `forEach()` this is avoided somehow... now I see that it's because the `forEach()` never runs (because `myOrgs` is null) and so the previous state of `this.listOfOrganizations` and `this.totalOrgs` is never reassigned.

https://user-images.githubusercontent.com/97123241/219804487-cd28d349-e7eb-4860-a597-197285c5cc44.mp4

I'll remove the `forEach()`, then addressing your previous comment, assigning `this.totalOrgs` correctly will prevent the card from going to `You are not in any organizations`.